.github/workflows/ci-build.yml

```diff
@@ -203,9 +203,10 @@ jobs:
 
     name: Mac OS (${{ matrix.arch }})
 
-    runs-on: macos-latest
+    runs-on: macos-13
     env:
       MACOSX_DEPLOYMENT_TARGET: 10.13
+      DEVELOPER_DIR: /Applications/Xcode_15.1.app/Contents/Developer
```

Comment:
Meh; this will be annoying to remember to remove at some point in time.

You now select 15.1, but 15 would also be sufficient already. 15.2 is released, but not in the image yet. In fact, it was scheduled for the 14th this month to be deployed.

Easy to forget about this, making less-optimal binaries.

On the other hand, `macos-latest` runs an even older XCode, so this is already an improvement :P So I guess I just have to get over the feeling of "meh".

Reply:
I used the path from https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md, and there I'm not seeing that /Applications/Xcode_15.app is a valid path.
Would that be possible? I got no clue. Lets try...

Reply:
No, Apple calls 15.0 just 15. And it got a security update, so it is called 15.0.1. I was reading Apple's docs when writing this comment :) There is no "15" to mean "latest in 15 series". There is just a default, which is 14.3.something.

Reply:
So lets conclude that `std::source_location` support is still "too new" in MacOS/Xcode to include it right now.
We'll have to wait until there's a `macos-latest` that defaults to Xcode 15 or higher. That might be with `macos-14` in about two years (13 is almost a year old, and still not the default for latest), but it might also be that in a few months 13 becomes latest, and the default Xcode becomes 15. It's hard to tell at this moment.

The only remaining question is, what to do with this PR. Currently it won't be merged, but having it open will likely prevent others from attempting the conversion and running into the same issue.

Reply:
For sure leave it open.

@glx22 what do you think? Do we merge this now, knowing we have to keep an eye on the XCode version? Or do we postpone till GitHub runners change the default?

PS: bit weird that they don't change the default for a beta-runner, but okay :)

Reply:
I think it's probably better to keep it open. Last time we hardcoded image version (it was macos-11 but I may be wrong) it was not changed until the warnings about close drop of the image started to be annoying.